extensions/azurecore/src/azureResource/services/subscriptionService.ts

```diff
@@ -31,18 +31,25 @@ export class AzureResourceSubscriptionService implements IAzureResourceSubscript
 		for (const tenantId of tenantIds ?? account.properties.tenants.map(t => t.id)) {
 			try {
 				const token = await azdata.accounts.getAccountSecurityToken(account, tenantId, azdata.AzureResource.ResourceManagement);
-				const subClient = new SubscriptionClient(new TokenCredentials(token.token, token.tokenType), { baseUri: account.properties.providerSettings.settings.armResource.endpoint });
-				const newSubs = await subClient.subscriptions.list();
-				subscriptions.push(...newSubs.map(newSub => {
-					return {
-						id: newSub.subscriptionId,
-						name: newSub.displayName,
-						tenant: tenantId
-					};
-				}));
-				gotSubscriptions = true;
+				if (token !== undefined) {
+					const subClient = new SubscriptionClient(new TokenCredentials(token.token, token.tokenType), { baseUri: account.properties.providerSettings.settings.armResource.endpoint });
+					const newSubs = await subClient.subscriptions.list();
+					subscriptions.push(...newSubs.map(newSub => {
+						return {
+							id: newSub.subscriptionId,
+							name: newSub.displayName,
+							tenant: tenantId
+						};
+					}));
+					gotSubscriptions = true;
+				}
+				else if (!account.isStale) {
```

Comment:
If the account is stale then shouldn't we be handling it in some better way - such as informing the user they need to re-authenticate or just starting that process right away? Although given the generic use of `getSubscriptions` and the like it seems like we'll want that to be controllable by the caller as well.

Reply:
In this case, `getSubscriptions` need not do that. User is prompted first time when they need to reauthenticate, and hence-forth they are directed to Azure Account window as in the video.

Reply:
Where is that re-prompting happening?

Reply:
Before marking connection as "Stale":
https://github.com/microsoft/azuredatastudio/blob/701f656c5542d06500a91f5cbd74f7f173e645ad/extensions/azurecore/src/account-provider/auths/azureAuth.ts#L226

Reply:
Hm, ok. I want to look more into where else getSubscriptions is used and whether things are correctly handled in those cases as well. But I'll do that separately so this seems fine for now.